packages/lesswrong/components/events/modules/VirtualProgramCard.tsx

```diff
@@ -0,0 +1,198 @@
+import moment from 'moment';
+import { useTracking } from '../../../lib/analyticsEvents';
+
+const styles = createStyles((theme: ThemeType): JssStyles => ({
+  eventCard: {
+    position: 'relative',
+    width: 373,
+    height: 374,
+    borderRadius: 0,
+    overflow: 'visible',
+    boxShadow: "0 1px 3px rgba(0, 0, 0, 0.1)",
+    [theme.breakpoints.down('xs')]: {
+      maxWidth: '100vw'
+    }
+  },
+  introVPCard: {
+    background: "linear-gradient(rgba(0, 87, 102, 0.7), rgba(0, 87, 102, 0.7)), url('https://res.cloudinary.com/cea/image/upload/w_374,h_373,c_fill,q_auto,f_auto/Event/pz3xmsm63xl8thlyt2up.jpg')",
+    padding: '50px 24px',
+    '& .VirtualProgramCard-eventCardDescription': {
+      opacity: 1,
+      lineHeight: '1.8em',
+      marginTop: 30
+    },
+    '& .VirtualProgramCard-eventCardDeadline': {
+      marginTop: 30
+    }
+  },
+  cardLink: {
+    '&:hover': {
+      opacity: '0.9'
+    },
+    '&:hover .VirtualProgramCard-eventCardDeadline': {
+      borderBottom: '2px solid white'
+    }
+  },
+  cardSection: {
+    display: 'flex',
+    width: 373,
+    height: 243,
+    padding: 20,
+    overflow: 'hidden'
+  },
+  inDepthSection: {
+    background: "linear-gradient(rgba(0, 87, 102, 0.7), rgba(0, 87, 102, 0.7)), url('https://res.cloudinary.com/cea/image/upload/w_374,h_243,c_fill,q_auto,f_auto/Event/f2cbeqvjyhyl6rhhzdsu.jpg')",
+    clipPath: 'polygon(0 0, 100% 0, 100% 54%, 0 100%)'
+  },
+  precipiceSection: {
+    background: "linear-gradient(rgb(168, 114, 51, 0.5), rgb(168, 114, 51, 0.5)), url('https://res.cloudinary.com/cea/image/upload/w_374,h_243,c_fill,q_auto,f_auto/Event/xfhrtorwdxxmplaofqa8.jpg')",
+    clipPath: 'polygon(0 46%, 100% 0, 100% 100%, 0 100%)',
+    position: 'absolute',
+    bottom: 0,
+    alignItems: 'flex-end',
+    textAlign: 'right'
+  },
+  eventCardTime: {
+    ...theme.typography.commentStyle,
+    fontSize: 14,
+    color: 'white'
+  },
+  eventCardTitle: {
+    ...theme.typography.headline,
+    color: 'white',
+    fontSize: 22,
+    marginTop: 8,
+    marginBottom: 0
+  },
+  eventCardLocation: {
+    ...theme.typography.commentStyle,
+    opacity: '0.7',
+    color: 'white',
+    fontSize: 14,
+    marginTop: 8,
+  },
+  eventCardDescription: {
+    ...theme.typography.commentStyle,
+    opacity: '0.7',
+    color: 'white',
+    fontSize: 14,
+    lineHeight: '1.5em',
+    marginTop: 10,
+  },
+  eventCardDeadline: {
+    ...theme.typography.commentStyle,
+    display: 'inline-block',
+    fontWeight: 'bold',
+    color: 'white',
+    fontSize: 16,
+    paddingBottom: 5,
+    marginTop: 10,
+    borderBottom: '2px solid transparent'
+  },
+}))
+
+
+const VirtualProgramCard = ({program, classes}: {
+  program: string,
+  classes: ClassesType,
+}) => {
+  const { captureEvent } = useTracking()
+  
+  // find the next deadline for applying to the Intro VP, which is the last Sunday of every month
+  let sunday = moment().day(0)
+  _.range(5).forEach(() => {
```

Comment:
```suggestion
    // Iterate through 5 sundays to find the one we want
  for (const sunday of _.range(5).map(x => moment().day(0).add(x, 'week'))) {
```

Reply:
Meta: My personal aesthetic really hates forEach`es, and think you probably would have written a break clause if you'd written a for-of loop here. I kinda want to suggest never using them.